src/python/WMComponent/AgentStatusWatcher/AgentStatusPoller.py

```diff
@@ -281,3 +283,32 @@ def checkProxyLifetime(self, agInfo):
             agInfo['proxy_warning'] = warnMsg
 
         return
+
+    def checkUserCertLifetime(self, agInfo):
```

Comment:
Kenyi, maybe we could reuse the same method as above (you can rename it to checkCredLifetime) and pass a parameter saying whether it's a proxy or certificate, which will be used for the error message too.

The threshold is wired in the code because I didn;t expect it to change too often, I think we can keep that. We could set something like at the very beginning
```
thresholds = {'proxy': {'error': 3, 'warn': 5},
                       'cert': {'error': 10, 'warn': 20}}
```

or something else that comes to your mind.
This line
`agInfo['proxy_warning'] = warnMsg`
has to become
`agInfo['proxy_warning'] += warnMsg`

to make sure we catch both messages in case both proxy and cert are about to expire.

Reply:
Will do, should I follow any convention of time units to show in debug? Right now, we are showing proxy in secs and cert in days.

Reply:
Done.